perun-core/src/test/java/cz/metacentrum/perun/core/entry/GroupsManagerEntryIntegrationTest.java

```diff
@@ -1450,6 +1451,61 @@ public void convertGroupsToRichGroupsWithAttributesWithListOfNamesTest() throws
 		assertEquals("Both lists should be same", Arrays.asList(richGroup1, richGroup2), groupsManagerBl.convertGroupsToRichGroupsWithAttributes(sess, Arrays.asList(group, group2), attrNames));
 	}
 
+	@Test(expected = ExternalyManagedException.class)
+	public void addMemberToSynchronizedGroup() throws Exception {
+		System.out.println(CLASS_NAME + "addMemberToSynchronizedGroup");
+
+		vo = setUpVo();
+		setUpGroup(vo);
+
+		Attribute synchroAttr = new Attribute(perun.getAttributesManager().getAttributeDefinition(sess, "urn:perun:group:attribute-def:def:synchronizationEnabled"));
+		synchroAttr.setValue("true");
+		perun.getAttributesManager().setAttribute(sess, group, synchroAttr);
+
+		Member member = setUpMember(vo);
+		groupsManager.addMember(sess, group, member);
+	}
+	
+	public void addAndRemoveMemberInNonSynchronizedGroup() throws Exception {
+		System.out.println(CLASS_NAME + "addAndRemoveMemberInNonSynchronizedGroup");
+
+		vo = setUpVo();
+		setUpGroup(vo);
+
+		Attribute synchroAttr = new Attribute(perun.getAttributesManager().getAttributeDefinition(sess, "urn:perun:group:attribute-def:def:synchronizationEnabled"));
+		synchroAttr.setValue("false");
+		perun.getAttributesManager().setAttribute(sess, group, synchroAttr);
+
+		Member member = setUpMember(vo);
+		groupsManager.addMember(sess, group, member);
+		List<Member> members = groupsManager.getGroupMembers(sess, group);
+		assertTrue("List of members should contain member", members.contains(member));
+		
+		groupsManager.removeMember(sess, group, member);
+		members.clear();
```

Comment:
You don't have to ``clear()`` list variable here since it's re-assigned on the next line -> all list content is replaced by the new assignment and old content is discarded by java garbage collector (if not referenced from anywhere else).